### PR TITLE
refactor: gains function `prepare_geco_2022_scenario()`

### PR DESCRIPTION
**Pull request by jdhoffa**
In the context of both [workflow.scenario.preparation](https://github.com/RMI-PACTA/workflow.scenario.preparation/blob/main/main.R) and this repo, this is a pure refactor.
In the context of this repo alone, this is a new feature.

* This package gains the exported function `prepare_geco_2022_scenario()`
* Function gains some simple testing
* All checks pass
* I explain how to run a regression test to see that no output has changed here https://github.com/RMI-PACTA/workflow.scenario.preparation/pull/15

Note: I also added some `fake_*` test input datasets to facilitate unit-testing
I am considering extracting those to https://github.com/RMI-PACTA/pacta.demo.data
I will make an issue to that effect after this is merged

Note: This is largely a proof of concept to be used for future scenario preparations. 
Since GECO 2022 is now two years old (and already prepared), I don't want to spend too much time thinking of so many tests or new warnings or errors for the function. However, I will absolutely push for a higher standard of acceptance for any future scenarios (2023 onward).

Relates to #12

### CODE REVIEW

File: tests/testthat/helper-fake_data.R [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    `2025` = `2025` %||% 54321
+  )
+}
+#' See `fake_technology_bridge()`
+#' @noRd
+fake_geco_2022_aviation_raw <- function(GECO = NULL,
+                                        Scenario = NULL,
+                                        Variable = NULL,
+                                        Sector = NULL,
+                                        `Passenger/Frieght` = NULL,
```
Comment (jacobvjk) [inline tests/testthat/helper-fake_data.R:60]: looks like a typo?
```suggestion
                                        `Passenger/Freight` = NULL,
```

**Review by jacobvjk — COMMENTED** [review]
seems to work well, one typo related comment

File: tests/testthat/helper-fake_data.R (hunk 1 above)
Comment (jdhoffa) [reply tests/testthat/helper-fake_data.R:60]: Thank you!

**Review by jacobvjk — APPROVED** [review]
lgtm
agree on comments regarding 2022 versus future scenario preparations

**Pull request merged**